.buildkite/x-pack/pipeline.xpack.heartbeat.yml

```diff
@@ -25,39 +25,189 @@ env:
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    - select: "Heartbeat - run_xpack_heartbeat_macos_tests"
-      key: "run_xpack_heartbeat_macos_tests"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-
-    if: "build.source == 'ui'"
+  #     - label: ":windows: Windows 11 Unit Tests"
+  #       command: |
+  #         Set-Location -Path $BEATS_PROJECT_NAME
+  #         mage build unitTest
+  #       key: "extended-win-11-unit-tests"
+  #       agents:
+  #         provider: "gcp"
+  #         image: "${IMAGE_WIN_11}"
+  #         machineType: "${GCP_WIN_MACHINE_TYPE}"
+  #         disk_size: 100
+  #         disk_type: "pd-ssd"
+  #       artifact_paths:
+  #         - "$BEATS_PROJECT_NAME/build/*.xml"
+  #         - "$BEATS_PROJECT_NAME/build/*.json"
+  #       notify:
+  #         - github_commit_status:
+  #             context: "$BEATS_PROJECT_NAME: Windows 11 Unit Tests"
+
+  #     - label: ":windows: Windows 2019 Unit Tests"
+  #       command: |
+  #         Set-Location -Path $BEATS_PROJECT_NAME
+  #         mage build unitTest
+  #       key: "extended-win-2019-unit-tests"
+  #       agents:
+  #         provider: "gcp"
+  #         image: "${IMAGE_WIN_2019}"
+  #         machineType: "${GCP_WIN_MACHINE_TYPE}"
+  #         disk_size: 100
+  #         disk_type: "pd-ssd"
+  #       artifact_paths:
+  #         - "$BEATS_PROJECT_NAME/build/*.xml"
+  #         - "$BEATS_PROJECT_NAME/build/*.json"
+  #       notify:
+  #         - github_commit_status:
+  #             context: "$BEATS_PROJECT_NAME: Windows 2019 Unit Tests"
+  ####### skip: "see elastic/beats#23957 and elastic/beats#23958"
+
+  - group: "MacOs Extended Tests"
+    key: "extended-tests-macos"
+    if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*macOS.*/
+    steps:
+      - label: ":mac: MacOS Unit Tests"
+        key: "extended-macos-unit-tests"
+        command: ".buildkite/scripts/unit_tests.sh"
+        agents:
+          provider: "orka"
+          imagePrefix: "${IMAGE_MACOS_X86_64}"
+        artifact_paths:
+          - "$BEATS_PROJECT_NAME/build/*.xml"
+          - "$BEATS_PROJECT_NAME/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "$BEATS_PROJECT_NAME: MacOS Unit Tests"
 
   - wait: ~
-    if: "build.source == 'ui'"
-    allow_dependency_failure: false
-
-  - label: ":linux: Load dynamic x-pack heartbeat pipeline"
-    key: "xpack-heartbeat-pipeline"
-    command: ".buildkite/scripts/generate_xpack_heartbeat_pipeline.sh"
-    agents:
-      image: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci:latest"
-    notify:
-      - github_commit_status:
-          context: "${BEATS_PROJECT_NAME}: Load dynamic pipeline's steps"
+    depends_on:
```

Comment:
we want to be able to run packaging regardless of mandatory/extended unit tests.
It's already ensured in line 184 below, however, in the context of a merge commit (e.g. in `main`) this wait, as it is now, will wait until all mandatory-tests have finished.

So let's just add if: build.env("BUILDKITE_PULL_REQUEST") != "false" above this line?

see https://github.com/elastic/beats/pull/38827/files#diff-87cc9e7aecd3c1a6382e951a1b17429443aa3f12387328521fa06b2c8f83a814R234-R237

for an example with comments